torch_em/model/unetr.py

```diff
@@ -167,25 +172,35 @@ def forward(self, x):
         # backbone used for reshaping inputs to the desired "encoder" shape
         x = torch.stack([self.preprocess(e) for e in x], dim=0)
 
-        z0 = self.z_inputs(x)
+        if self.use_skip_connection:
+            z0 = self.z_inputs(x)
 
         z12, from_encoder = self.encoder(x)
-        x = self.base(z12)
 
-        from_encoder = from_encoder[::-1]
-        z9 = self.deconv1(from_encoder[0])
+        if self.use_skip_connection:
+            from_encoder = from_encoder[::-1]
+            z9 = self.deconv1(from_encoder[0])
 
-        z6 = self.deconv1(from_encoder[1])
-        z6 = self.deconv2(z6)
+            z6 = self.deconv1(from_encoder[1])
+            z6 = self.deconv2(z6)
 
-        z3 = self.deconv1(from_encoder[2])
-        z3 = self.deconv2(z3)
-        z3 = self.deconv3(z3)
+            z3 = self.deconv1(from_encoder[2])
+            z3 = self.deconv2(z3)
+            z3 = self.deconv3(z3)
```

Comment:
I don't find it intuitive that we share the weights of the deconvs here. But since this is in the previous part of the UNETR implementation this should be fine.

Reply:
Added a comment on this. Thanks!